openedx/core/djangoapps/ccxcon/api.py

```diff
@@ -0,0 +1,146 @@
+"""
+Module containing API functions for the CCXCon
+"""
+
+import logging
+import urlparse
+from oauthlib.oauth2 import BackendApplicationClient
+from requests_oauthlib import OAuth2Session
+from rest_framework.status import HTTP_201_CREATED
+
+from django.core.validators import URLValidator
+from django.core.exceptions import ValidationError
+from django.http import Http404
+
+from courseware.courses import get_course_by_id
+from instructor.access import list_with_level
+from openedx.core.djangoapps.models.course_details import CourseDetails
+from student.models import anonymous_id_for_user
+from .models import CCXCon
+
+log = logging.getLogger(__name__)
+
+
+CCXCON_COURSEXS_URL = '/api/v1/coursexs/'
+
+
+class CCXConnServerError(Exception):
+    """
+    Custom exception to be raised in case there is any
+    issue with the request to the server
+    """
+
+
+def is_valid_url(url):
+    """
+    Helper function used to check if a string is a valid url.
+
+    Args:
+        url (str): the url string to be validated
+
+    Returns:
+        bool: whether the url is valid or not
+    """
+    validate = URLValidator()
+    try:
+        validate(url)
+        return True
+    except ValidationError:
+        return False
+
+
+def get_oauth_client(server_token_url, client_id, client_secret):
+    """
+    Function that creates an oauth client and fetches a token.
+    It intentionally doesn't handle errors.
+
+    Args:
+        server_token_url (str): server URL where to get an authentication token
+        client_id (str): oauth client ID
+        client_secret (str): oauth client secret
+
+    Returns:
+        an instance of OAuth2Session with a token
+    """
+    if not is_valid_url(server_token_url):
+        return
+    client = BackendApplicationClient(client_id=client_id)
+    oauth_ccxcon = OAuth2Session(client=client)
+    oauth_ccxcon.fetch_token(
+        token_url=server_token_url,
+        client_id=client_id,
+        client_secret=client_secret,
+        verify=False
```

Comment:
In production we will have a valid certificate so we should remove this line. It may have been `verify=False` in development due to a self signed certificate